packages/flutter_test/lib/src/binding.dart

```diff
@@ -1776,3 +1779,44 @@ class _MockHttpHeaders extends HttpHeaders {
   @override
   String value(String name) => null;
 }
+
+/// Helper class that mocks assets message channel if platform environment
+/// variable exists.
+class _AssetMocker {
+  _AssetMocker();
+
+  static Set<dynamic> _allowedKeys;
+
+  /// Mocks message channel.
+  ///
+  /// This function will mock message channel or override existing mock every
+  /// time it is called.
+  static void mock() {
```

Comment:
I updated this function. Previously, it only mock when the first time it is called, and no op at later call.
That will pose an issue when some of our tests also mock the message channel for some specific test cases.

I updated it so it will mock or override existing mock every time it is called. To ensure the performance, I implement _ensureInitialized method to do the expansive operation.